File: services/ingestion-openmeteo/app/Domain/Weather/Location.php
<?php

namespace App\Domain\Weather;

/**
 * Domain entitet koji predstavlja lokaciju za koju prikupljamo vremenske podatke.
 */
final readonly class Location
{
    /**
     * Kreira novu lokaciju.
     *
     * @param string $id
     * @param string $name
     * @param string $country
     * @param float $latitude
     * @param float $longitude
     * @param bool $isActive
     */
    public function __construct(
        public string $id,
        public string $name,
        public string $country,
        public float $latitude,
        public float $longitude,
        public bool $isActive = true,
    ) {
    }

    /**
     * Vraća lokaciju kao niz podataka.
     *
     * @return array
     */
    public function toArray(): array
    {
        return [
            'id' => $this->id,
            'name' => $this->name,
            'country' => $this->country,
            'latitude' => $this->latitude,
            'longitude' => $this->longitude,
            'is_active' => $this->isActive,
        ];
    }
}

File: services/ingestion-weatherapi/internal/domain/weather/location.go
package weather

/*
Location predstavlja domen entitet lokacije za koju prikupljamo vremenske podatke.
*/
type Location struct {
	ID        string
	Name      string
	Country   string
	Latitude  float64
	Longitude float64
	IsActive  bool
}

/*
NewLocation pravi novu instancu Location entiteta.
*/
func NewLocation(
	id string,
	name string,
	country string,
	latitude float64,
	longitude float64,
	isActive bool,
) Location {
	return Location{
		ID:        id,
		Name:      name,
		Country:   country,
		Latitude:  latitude,
		Longitude: longitude,
		IsActive:  isActive,
	}
}
